bin/codemods/src/assign-component-to-displayname-variable.js

```diff
@@ -50,91 +39,73 @@ const displayNamePropertyIdentifier = {
-	arg.get( 'callee', 'property', 'name' ).value === 'createClass';
-
-const argIsExtendsComponentInstance = arg =>
-	arg.get( 'type' ).value === 'ClassExpression' &&
-	( isValidComponentString( arg.get( 'superClass', 'name' ).value ) ||
-		( arg.get( 'superClass', 'object', 'name' ).value === 'React' &&
-			isValidComponentString( arg.get( 'superClass', 'property', 'name' ).value ) ) );
-
-const argIsComponent = arg =>
-	argIsCreateClassInstance( arg ) || argIsExtendsComponentInstance( arg );
-
-const getMatchingArg = args => args.filter( argIsComponent )[ 0 ];
+const isCreateClassComponent = arg =>
+	_.get( arg, [ 'callee', 'object', 'name' ] ) === 'React' &&
+	_.get( arg, [ 'callee', 'property', 'name' ] ) === 'createClass';
+
+const isEs6ClassComponent = arg =>
+	_.get( arg, 'type' ) === 'ClassExpression' &&
+	( isValidComponentString( _.get( arg, [ 'superClass', 'name' ] ) ) ||
+		( _.get( arg, [ 'superClass', 'object', 'name' ] ) === 'React' &&
+			isValidComponentString( _.get( arg, [ 'superClass', 'property', 'name' ] ) ) ) );
+
+const isComponent = arg => isCreateClassComponent( arg ) || isEs6ClassComponent( arg );
+
+const getComponentFromArgs = args => _.filter( args, isComponent )[ 0 ];
+
+const classToIdentifier = ( displayName, j ) => arg =>
+	isComponent( arg ) ? j.identifier( displayName ) : arg;
+
+const extractDisplayName = ( reactComponent, j ) => {
+	if ( ! reactComponent ) {
+		return;
+	}
+	const component = j( reactComponent );
+	// can work on both es6 classes and React.createClass functions.
+	return []
+		.concat(
+			component.find( j.Property, displayNamePropertyIdentifier ).nodes(),
+			component.find( j.ClassProperty, displayNamePropertyIdentifier ).nodes()
+		)
+		.map( node => _.get( node, 'value.value' ) )[ 0 ];
+};
 
 export default function transformer( file, api ) {
 	const j = api.jscodeshift;
 	const root = j( file.source );
 
-	// string => object =>
-	const replaceClassOrGetValue = displayName => arg =>
-		argIsComponent( arg ) ? j.identifier( displayName ) : arg.value;
-
-	const exportDefaultDeclarations = root.find( j.ExportDefaultDeclaration );
-
-	exportDefaultDeclarations.forEach( node => {
-		const calleeNameValue = node.get( 'declaration', 'callee', 'name' ).value;
+	const defaultExportDeclaration = _.head( root.find( j.ExportDefaultDeclaration ).nodes() );
+	const hocIdentifier = _.get( defaultExportDeclaration, [ 'declaration', 'callee', 'name' ] );
 
-		if ( ! calleeNameValue ) {
-			// return early, no immediate function call
-			return;
-		}
+	const hocArgs = _.get( defaultExportDeclaration, [ 'declaration', 'arguments' ] );
+	const component = getComponentFromArgs( hocArgs );
+	const displayName = extractDisplayName( component, j );
 
-		const exportDefaultInstance = j( node );
+	// noop if the file does not have a default export of a react component that has a displayName
+	if ( ! defaultExportDeclaration || ! component || ! displayName ) {
+		return file;
```

Comment:
Aha, this needs to be `file.source` to avoid rewriting all the would-be-skipped files to `[Object object]` :)

Reply:
I think just `return` should work as well, i'll investigate.